libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -187,56 +187,6 @@ method onNewPeer(g: GossipSub, peer: PubSubPeer) =
     peer.appScore = stats.appScore
     peer.behaviourPenalty = stats.behaviourPenalty
 
-proc grafted(g: GossipSub, p: PubSubPeer, topic: string) =
-  g.peerStats.withValue(p.peerId, stats):
-    var info = stats.topicInfos.getOrDefault(topic)
-    info.graftTime = Moment.now()
-    info.meshTime = 0.seconds
-    info.inMesh = true
-    info.meshMessageDeliveriesActive = false
-
-    # mgetOrPut does not work, so we gotta do this without referencing
-    stats.topicInfos[topic] = info
-    assert(g.peerStats[p.peerId].topicInfos[topic].inMesh == true)
-
-    trace "grafted", peer=p, topic
-  do:
-    g.initPeerStats(p)
-    g.grafted(p, topic)
-
-proc pruned(g: GossipSub, p: PubSubPeer, topic: string) =
-  let backoff = Moment.fromNow(g.parameters.pruneBackoff)
-  g.backingOff
-    .mgetOrPut(topic, initTable[PeerID, Moment]())
-    .mgetOrPut(p.peerId, backoff) = backoff
-
-  g.peerStats.withValue(p.peerId, stats):
-    if topic in stats.topicInfos:
-      var info = stats.topicInfos[topic]
-      if topic in g.topicParams:
-        let topicParams = g.topicParams[topic]
-        # penalize a peer that delivered no message
-        let threshold = topicParams.meshMessageDeliveriesThreshold
-        if info.inMesh and info.meshMessageDeliveriesActive and info.meshMessageDeliveries < threshold:
-          let deficit = threshold - info.meshMessageDeliveries
-          info.meshFailurePenalty += deficit * deficit
-
-      info.inMesh = false
-
-      # mgetOrPut does not work, so we gotta do this without referencing
-      stats.topicInfos[topic] = info
-
-      trace "pruned", peer=p, topic
-
-proc peerExchangeList(g: GossipSub, topic: string): seq[PeerInfoMsg] =
-  var peers = g.gossipsub.getOrDefault(topic, initHashSet[PubSubPeer]()).toSeq()
-  peers.keepIf do (x: PubSubPeer) -> bool:
-      x.score >= 0.0
-  # by spec, larger then Dhi, but let's put some hard caps
-  peers.setLen(min(peers.len, g.parameters.dHigh * 2))
-  peers.map do (x: PubSubPeer) -> PeerInfoMsg:
-    PeerInfoMsg(peerID: x.peerId.getBytes())
-
 proc replenishFanout(g: GossipSub, topic: string) =
```

Comment:
Any reason to leave this method here - seems like it would make sense to group it in `behaviour.nim`?